cybox/common/vocabs.py

```diff
@@ -6,13 +6,58 @@
 from cybox.common import PatternFieldGroup
 from cybox.utils import normalize_to_xml, denormalize_from_xml
 
-INVALID_XSI_TYPE = "bad_xsi_type"
+
+class VocabField(cybox.TypedField):
+    """TypedField subclass for VocabString fields.
+
+    """
+    def __init__(self, *args, **kwargs):
+        """Intercepts the __init__() call to TypedField.
+
+        Set the type that will be used in from_dict and from_obj calls to
+        :class:`VocabString`.
+
+        Set the type that will be used in ``__set__`` for casting as the
+        original ``type_`` argument, or :class:`VocabString` if no `type_`
+        argument was provided.
+
+        """
+        super(VocabField, self).__init__(*args, **kwargs)
+
+        if self.type_:
+            self.__vocab_impl = self.type_
+        else:
+            self.__vocab_impl = VocabString
+
+        self.type_ = VocabString  # Force this
+
+    def __set__(self, instance, value):
+        """Overrides cybox.TypedField.__set__()."""
+        type_ = self.__vocab_impl
```

Comment:
I would rename this variable, since it's different than `self.type_`. Maybe just `vocab`?